### ISSUE TITLE
QPR and Burst setting for Kubernetes client

### ISSUE THREAD

**Issue by chanwit** [weaveworks/weave-gitops#2268]
Weave GitOps got many connection errors because the client's configuration does not have a rate limiter.

```
I0603 09:54:37.501060       1 request.go:665] Waited for 1.036691978s due to client-side throttling, not priority and fairness, request: GET:[https://10.60.0.1:443/apis/authentication.k8s.io/v1?timeout=32s](https://10.60.0.1/apis/authentication.k8s.io/v1?timeout=32s)
2022-06-03T13:17:07.063Z	INFO	gitops	middleware/middleware.go:61	request error	{"uri": "/v1/flux_runtime_objects?clusterName=Default", "status": 401}
2022-06-03T13:17:07.260Z	INFO	gitops	middleware/middleware.go:61	request error	{"uri": "/oauth2/userinfo", "status": 400}
I0603 13:17:08.259404       1 request.go:665] Waited for 1.043454555s due to client-side throttling, not priority and fairness, request: GET:[https://10.60.0.1:443/apis/helm.toolkit.fluxcd.io/v2beta1?timeout=32s](https://10.60.0.1/apis/helm.toolkit.fluxcd.io/v2beta1?timeout=32s)
```

We should set QPR and Burst values like the following for our Kube client.
```
	config := ctrl.GetConfigOrDie()
	enabled, err := flowcontrol.IsEnabled(context.Background(), config)
	if err == nil && enabled {
		// A negative QPS and Burst indicates that the client should not have a rate limiter.
		// Ref: https://github.com/kubernetes/kubernetes/blob/v1.24.0/staging/src/k8s.io/client-go/rest/config.go#L354-L364
		config.QPS = -1
		config.Burst = -1
		return config
	}
	config.QPS = opts.QPS
	config.Burst = opts.Burst
	return config
```

**Comment by chanwit** [comment]
A Flux controller uses QPS=50 and Burst=100 at the moment.
While WEGO uses QPS=1000 and Burst=2000.

What would be the good values for applications like WEGO?

**Comment by JamWils** [comment]
This is on hold for Kubernetes upstream library (kube-openapi) to merge a change.  Once that is done this can be reviewed and merged.

**Comment by chanwit** [comment]
The CVE that blocked this card has now been published. It's CVE-2022-1996.
We can continue to work on this card.
